gdb-server/src/writer.rs

```diff
@@ -1,58 +1,57 @@
 use async_std::{io::Write, net::TcpStream, prelude::*};
 use futures::channel::mpsc;
 use gdb_protocol::packet::{CheckedPacket, Kind as PacketKind};
-use std::sync::Arc;
 
 type Result<T> = std::result::Result<T, Box<dyn std::error::Error + Send + Sync>>;
 type Sender<T> = mpsc::UnboundedSender<T>;
 
 pub async fn writer(
     packet: CheckedPacket,
-    stream: Arc<TcpStream>,
-    packet_stream: Sender<CheckedPacket>,
```

Comment:
Why make the Sender a ref? The Sender is meant to be owned afaik, as the mpsc can produce as many as it wants.

Reply:
Making a clone of the sender for every packet seems very wasteful, when it's not necessary. Right now I don't see an advantage when this is owned by the function.